pkg/build/builder/docker.go

```diff
@@ -413,6 +419,33 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV Dockerfile instruction right after the
+// FROM instruction of the given dockerfile to define the environment variables
+// from env and returns the new Dockerfile.
+func insertEnvAfterFrom(env []kapi.EnvVar, dockerfile []byte) []byte {
+	if len(env) == 0 {
+		// Just return a copy.
+		return append([]byte(nil), dockerfile...)
+	}
+	fromLine := dockerfileFromLineRegexp.Find(dockerfile)
```

Comment:
@rhcarvalho this is already solved here: https://github.com/openshift/origin/blob/5802c03d39de62ca92c20719835cf2c209891856/pkg/build/builder/docker.go#L386

You just have to parse the dockerfile and call `traverseAST("FROM", root)`. Since you want to edit the next line and not FROM you will have to refactor that code a bit to be more re-usable.